### ISSUE TITLE
Fix error reporting

### ISSUE THREAD

**Issue by raquelxmoss** [Widdershin/tricycle#6]
**Expected behaviour:**
Removing `DOM: Observable.just(div('.hello-world', 'Hello world!'))` and returning an empty function renders a blank results container.

**Actual behaviour:**
Removing `DOM: Observable.just(div('.hello-world', 'Hello world!'))` continues to display "Hello World" in the results container.

**To replicate:**
Write an empty function in the editor. 

<img width="718" alt="screen shot 2015-11-26 at 8 50 51 am" src="https://cloud.githubusercontent.com/assets/7959274/11408346/548e9d78-941d-11e5-85fb-92ad84009d14.png">

**Comment by Widdershin** [comment]
Yeah, I think that's because we broke the error reporting last night, and if you don't return an object with a key of DOM and a DOM observable stream as the value, it doesn't work.

**Comment by Widdershin** [comment]
I think this is actually about errors that start up after the application is run but don't happen synchronously. We would need something like a `.catch` in every driver's stream to get all the errors. Something to talk to staltz about perhaps
